subprojects/test-kit/src/testFixtures/groovy/org/gradle/testkit/runner/BaseGradleRunnerIntegrationTest.groovy

```diff
@@ -180,31 +184,32 @@ abstract class BaseGradleRunnerIntegrationTest extends AbstractIntegrationSpec {
         OutputScrapingExecutionFailure.from(buildResult.output, buildResult.output)
     }
 
-    static String determineMinimumVersionThatRunsOnCurrentJavaVersion(String desiredGradleVersion) {
-        if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_17)) {
-            def compatibleVersion = GradleVersion.version("7.3-rc-2")
-            if (GradleVersion.version(desiredGradleVersion) < compatibleVersion) {
-                return compatibleVersion.version
-            }
-        }
-        if (JavaVersion.current().isCompatibleWith(JavaVersion.VERSION_16)) {
-            def compatibleVersion = GradleVersion.version("7.0")
-            if (GradleVersion.version(desiredGradleVersion) < compatibleVersion) {
-                return compatibleVersion.version
-            }
-        }
-        if (JavaVersion.current().isJava11Compatible()) {
-            def compatibleVersion = GradleVersion.version("4.8.1") // see https://github.com/gradle/gradle/issues/4860
-            if (GradleVersion.version(desiredGradleVersion) < compatibleVersion) {
-                return compatibleVersion.version
-            }
-        } else if (JavaVersion.current().isJava9Compatible()) {
-            def compatibleVersion = GradleVersion.version("4.3.1") // see https://github.com/gradle/gradle/issues/2992
-            if (GradleVersion.version(desiredGradleVersion) < compatibleVersion) {
-                return compatibleVersion.version
-            }
+    // Add to this after adding support for a JDK version.
+    // If the existing version still works, increase its range to include the next version.
+    // If it doesn't, add a new one at the top with the release/rc version.
+    static final def GRADLE_VERSION_BY_JDK
```

Comment:
We have a similar thing going on in `org.gradle.integtests.fixtures.executer.DefaultGradleDistribution#doesWorkWith`, can you see if we can re-use that here or at least update both places?

Reply:
I made an attempt at this, I don't know if it's really better than what already existed.